lib/logstash/inputs/redis.rb

```diff
@@ -23,12 +23,19 @@ module LogStash module Inputs class Redis < LogStash::Inputs::Threadable
 
   default :codec, "json"
 
-  # The hostname of your Redis server.
-  config :host, :validate => :string, :default => "127.0.0.1"
+  # The hostname of your Redis server
+  # This defaults to 127.0.0.1 if not specified.
+  # This and :path are mutually exclusive. Will raise an ArgumentError if both are specified.
+  config :host, :validate => :string
```

Comment:
I think we can keep this default here, and simple have the `:path` option overwrite host/port like described in [line 34](https://github.com/logstash-plugins/logstash-input-redis/pull/64/files#diff-e5d5a1b4209a0709fa9d7d9d29c28bf8R34)

Reply:
If `:path` is set we can just log a warn saying that path is set, so we're ignoring host/port

Reply:
Thanks @jsvd for the feedback. Ironically, that was actually the original logic of how this PR was constructed (see commit: https://github.com/logstash-plugins/logstash-input-redis/pull/64/commits/7c29668886b51f76bc64db949337afd8b7b2e48b) Looks like I forgot to remove the comment on L34 in the subsequent commits 😆 . 

@jordansissel - thoughts? I can see merits for both logic paths: 

1. Mutually exclusive
- No ambiguity about which connection is being used
- Defaults are no longer set at init, which does make the code more complex

2. Override
- Could potentially lead to confusion (especially if someone doesn't know what unix sockets are). Documentation can help assay this issue, but it's not as explicit as a mutex. 
- Defaults are set at init, making the code cleaner.